### ISSUE TITLE
Latex incorrectly triggering header-related violations

### ISSUE THREAD

**Issue by pancakereport** [berkeley-dsep-infra/jupyterlab-a11y-checker#48]
I'm having a hard time understanding what's going on here. I _think_ the checker is somehow interpreting some latex as a h1 header (heading-multiple-h1 violation), which ends up triggering another violation (heading-wrong-order) because it sees the h1 ahead of an h3, even though there really is a h2 before it. 

[Action output](https://github.com/DS-100/sp26-dev/actions/runs/24370595229/job/71173128210#step:3:73):
```
Found 2 issues in lec/lec12/lec12.ipynb:

1 violation found for heading-multiple-h1:
    Description: Ensure there is only one H1 tag in a notebook
  - Cell 6 (markdown):
    Content: "$$ \hat{y} ="

1 violation found for heading-wrong-order:
    Description: Ensure the order of heading is accurate
  - Cell 13 (markdown):
    Content: "### Example prediction"
```

Notebook: https://github.com/DS-100/sp26-dev/blob/a93c034a0a0f4544d90f2ebd13263124eb90054e/lec/lec12/lec12.ipynb

**Comment by pancakereport** [comment]
I tried putting the latex all on one line and that results in the checker not flagging any violations. I will make the change to our notebook so the violation isn't raised, but it would be nice if this could be addressed. I'm not exactly sure what part of the multi-line latex is causing issues (because I'm sure we've got other examples of multi-line latex that aren't being flagged as h1).

old (violations flagged):
```
$$
\hat{y}
=
\theta_0 +
\theta_1 \cdot \textbf{FG} +
\theta_2 \cdot \textbf{AST} +
\theta_3 \cdot \textbf{3PA}
$$
```
updated (no violations flagged):
```
$$ \hat{y} = \theta_0 + \theta_1 \cdot \textbf{FG} + \theta_2 \cdot \textbf{AST} + \theta_3 \cdot \textbf{3PA} $$
```

**Comment by pancakereport** [comment]
public notebook (with violations flagged): https://github.com/DS-100/sp26-student/blob/8b96e85e58a3e9eefdcf23ab7b4dd1973f25e5d8/lec/lec12/lec12.ipynb

notebook should update tomorrow around noon with the fix that doesn't cause a11y violations (above link should be stable to the commit with violations flagged)

**Comment by Chanbinski2** [comment]
Thanks for the issue.

Root cause: The standalone = line inside your multi-line $$...$$ block was being interpreted by the markdown parser as a [setext heading](https://spec.commonmark.org/0.31.2/#setext-headings) (where text underlined with = becomes an H1). This created a phantom H1 from \hat{y}, which then cascaded into both "Duplicate H1 Heading" and "Wrong Heading Order" violations.

Fix: We now mask $$...$$ math blocks before heading detection so the parser never sees the stray =. 

I've updated the extension on pypi and cli on npm. Once @balajialg updates the a11y datahub, you should be able to use it!

**Comment by pancakereport** [comment]
Yay thank you!
